src/main/scala/mesosphere/marathon/core/task/update/impl/steps/NotifyHealthCheckManagerStepImpl.scala

```diff
@@ -14,6 +14,7 @@ import scala.concurrent.Future
 class NotifyHealthCheckManagerStepImpl @Inject() (healthCheckManagerProvider: Provider[HealthCheckManager])
   extends InstanceChangeHandler {
   override def name: String = "notifyHealthCheckManager"
+  override def metricName: String = "notify-health-check-manager"
```

Comment:
surprised we did create a function to convert camelCase -> camel-case . 
not a fan of inheritance for state...  it would be better to have a class class for StepName with a name and metric IMO

Reply:
I don't think I understand what you mean here. Could you please elaborate?

Reply:
all metricNames are just conversions of the name of the step...  `notifyHealthCheckManager` -> `notify-health-check-manager` . all the steps follow this pattern.   It would be great to use a function that provides that... that way we have that behavior guaranteed.   It isn't a big deal.